### ISSUE TITLE
SScircuit_component sometimes drops callbacks

### ISSUE THREAD

**Issue by Y0SH1M4S73R** [tgstation/tgstation#62615]
Found on local server

## Reproduction:

- Load [this circuit](https://pastebin.com/tR2wHYJ8) (the pastebin will expire 6 months from the day this issue was opened)
- Insert it into an organ with a shell component (BCIs count for this purpose)
- Insert/Remove the organ by any means available
- One of the signals doesn't execute the corresponding speech component, even if the speech cooldown has expired

I don't know if the underlying issue is with the way the circuit component subsystem executes its callbacks, or if I ran into an intentional limitation of circuits.

**Comment by timothymtorres** [comment]
I don't know anything about circuits but maybe there is a hardlimit to the amount of code you can add to something.

**Comment by Mooshimi** [comment]
@Watermelon914 no knowledge of this send help

**Comment by Watermelon914** [comment]
I think I know why this is happening

**Comment by Watermelon914** [comment]
Your circuit doesn't work because you are

**Comment by Watermelon914** [comment]
misclick

**Comment by Watermelon914** [comment]
After doing some further testing, callbacks aren't getting deleted, there's no issues with any sort of callbacks. The speech components are getting triggered and they are calling shell.say, the shell just isn't saying anything when inserted because it is in a null location right before it gets inserted. It does say 'removed' because it is in the implanter by the time it gets removed.
![image](https://user-images.githubusercontent.com/37270891/146636926-62c883b0-9f14-4d15-962c-1e405404c539.png)

In conclusion, I don't think there is any issue with circuit code right now since it's all working as intended. You may want to take another look at your circuit

**Comment by Mooshimi** [comment]
> In conclusion, I don't think there is any issue with circuit code right now since it's all working as intended. You may want to take another look at your circuit

Thanks for your quick reply!
